### PR TITLE
fix: segmented control pass event to onChange, fix button tests

### PR DESCRIPTION
**Pull request by Giogia**
##### [Button]

Added mouse event type to onClick, fixed tests 

##### [Segmented Control]

Added mouse event parameter to onChange function

### Checklist

- [ ] added the link to the Jira task
- [x] commit message and branch name follow conventions
- [x] tests are included
- [x] changes are accessible and documented from components stories
- [x] typings are updated or integrated accordingly with your changes
- [x] all added files include Apache 2.0 license
- [ ] the changelog has been updated accordingly with your changes
- [x] you are not committing extraneous files or sensible data
- [x] the browser console does not have any logged errors

### CODE REVIEW

**Review by RiccardoCorona-MiaPlatform — DISMISSED** [review]

File: src/components/Button/Button.test.tsx [hunk 1]
```diff
@@ -136,5 +136,12 @@ describe('Button Component', () => {
     const button = screen.getByRole('button', { name: /Button/i })
     fireEvent.click(button)
     expect(onClick).toHaveBeenCalledTimes(1)
+    expect(onClick).toBeCalledWith(expect.objectContaining({
```
Comment (davidebianchi) [inline src/components/Button/Button.test.tsx:139]: It is needed to add this test? We should expect a MouseEvent maybe?

File: src/components/SegmentedControl/SegmentedControl.test.tsx [hunk 2]
```diff
@@ -136,7 +136,13 @@ describe('Segmented Control Component', () => {
       fireEvent.click(screen.getByRole('listitem', { name: clickedOption.key }))
 
       expect(onChange).toBeCalledTimes(1)
-      expect(onChange).toBeCalledWith(clickedOption)
+      expect(onChange).toBeCalledWith(clickedOption, expect.objectContaining({
```
Comment (davidebianchi) [inline src/components/SegmentedControl/SegmentedControl.test.tsx:139]: Also here

**Review by davidebianchi — CHANGES_REQUESTED** [review]

**Review by Danielecina — DISMISSED** [review]

File: src/components/Button/Button.test.tsx (hunk 1 above)
Comment (Giogia) [reply src/components/Button/Button.test.tsx:139]: I tried using expect.any(MouseEvent) but it fails. 
This to me was the closest way to test what we need from a MouseEvent. 

Would you have any recommendations on how to improve it?

Comment (Giogia) [reply src/components/Button/Button.test.tsx]: we swtiched to expect(onChange).toBeCalledWith(expect.objectContaining({ ...MouseEvent })), still not the best but should be closer to what we expect

File: src/components/SegmentedControl/SegmentedControl.test.tsx (hunk 2 above)
Comment (Giogia) [reply src/components/SegmentedControl/SegmentedControl.test.tsx]: Same approach adopted in the other comment

**Pull request merged**